src/snapshot-editor/src/info.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::path::PathBuf;
+
+use clap::Subcommand;
+use vmm::persist::MicrovmState;
+
+use crate::utils::*;
+
+#[derive(Debug, thiserror::Error)]
+pub enum InfoVmStateError {
+    #[error("{0}")]
+    Utils(#[from] UtilsError),
+}
+
+#[derive(Debug, Subcommand)]
+pub enum InfoVmStateSubCommand {
+    /// Print info about vcpu states.
+    #[cfg(target_arch = "aarch64")]
+    VcpuStates {
+        /// Path to the vmstate file.
+        #[arg(short, long)]
+        vmstate_path: PathBuf,
+    },
+}
+
+pub fn info_vmstate_command(command: InfoVmStateSubCommand) -> Result<(), InfoVmStateError> {
+    match command {
+        #[cfg(target_arch = "aarch64")]
+        InfoVmStateSubCommand::VcpuStates { vmstate_path } => {
+            info(&vmstate_path, info_vcpu_states)?
+        }
+    }
+    Ok(())
+}
+
+fn info(
+    vmstate_path: &PathBuf,
+    f: impl Fn(&MicrovmState, u16) -> Result<(), InfoVmStateError>,
```

Comment:
Mh, why does this closure take a `u16` for the version, while the one of the `edit` function in `edit_vmstate.rs` does not? Especially since the field is just ignored below

Reply:
initially the tool could also tell the snapshot/firecracker version, but I removed it to make PR smaller.
Removed the `u16` parameter.